tests/common.go

```diff
@@ -1356,6 +1366,50 @@ func GetAppNamespace(ctx *scheduler.Context, taskname string) string {
 	return ctx.App.GetID(fmt.Sprintf("%s-%s", taskname, Inst().InstanceID))
 }
 
+// Using the cuurent Context (kubeconfig) to generate schedule options
+// NOTE: When using a ScheduleOption that was created during a context (kubeconfig)
+// that is different from the current context, make sure to re-generate ScheduleOptions
+func CreateScheduleOptions(errChan ...*chan error) scheduler.ScheduleOptions {
+	var err error
+	options := scheduler.ScheduleOptions{
+		AppKeys:            Inst().AppList,
+		StorageProvisioner: Inst().Provisioner,
+	}
+
+	log.Infof("Creating ScheduleOptions")
+
+	//if not hyper converged set up deploy apps only on storageless nodes
+	if !Inst().IsHyperConverged {
+		log.Infof("ScheduleOptions: Scheduling apps only on storageless nodes")
+		storagelessNodes := node.GetStorageLessNodes()
+		if len(storagelessNodes) == 0 {
+			log.Info("ScheduleOptions: No storageless nodes available in the PX Cluster. Setting HyperConverges as true")
+			Inst().IsHyperConverged = true
+		}
+		for _, storagelessNode := range storagelessNodes {
+			if err = Inst().S.AddLabelOnNode(storagelessNode, "storage", "NO"); err != nil {
+				err = fmt.Errorf("ScheduleOptions: failed to add label key [%s] and value [%s] in node [%s]. Error:[%v]",
+					"storage", "NO", storagelessNode.Name, err)
+				processError(err, errChan...)
+			}
+		}
+		storageLessNodeLabels := make(map[string]string)
+		storageLessNodeLabels["storage"] = "NO"
+
+		options = scheduler.ScheduleOptions{
+			AppKeys:            Inst().AppList,
+			StorageProvisioner: Inst().Provisioner,
+			Nodes:              storagelessNodes,
+			Labels:             storageLessNodeLabels,
+		}
+
+	} else {
```

Comment:
To improve readability, it might be helpful to create two separate ScheduleOptions variables depending on the cluster configuration, instead of updating the same variable in both the if and else blocks.